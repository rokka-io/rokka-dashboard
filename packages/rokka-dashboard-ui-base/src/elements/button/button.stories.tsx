import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Button } from './button';

storiesOf('02 - Elements', module).add('Button', () => (
  <Button onClick={action('Button Click')}>{text('text', 'foobar')}</Button>
));
