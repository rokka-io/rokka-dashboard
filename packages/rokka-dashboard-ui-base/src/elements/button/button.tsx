import React, { FunctionComponent, ReactNode } from 'react';
import styled from 'styled-components';
import { colors } from '../../identity/colors/colors';
import { spaces } from '../../identity/spaces/spaces';
import { fonts } from '../../identity/typography/index';

export interface ButtonProps {
  /** Children */
  children?: ReactNode;
  /** Button type */
  type?: string;
  /** Whether button is disabled */
  disabled?: boolean;
  /** Callback */
  onClick?(): void;
}

export const Button: FunctionComponent<ButtonProps> = ({ children, type = 'button', onClick, disabled = false }) => (
  <StyledButton type={type} onClick={disabled ? undefined : onClick} disabled={disabled}>
    {children}
  </StyledButton>
);

interface StyledButtonProps {
  disabled?: boolean;
}

const StyledButton = styled.button<StyledButtonProps>`
  background-color: ${colors.brand.primary};
  color: ${colors.tints.white};
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  font-family: ${fonts.Families.semiBold};
  font-size: ${fonts.Sizes.medium};
  padding: 0 ${spaces.large};
  border: 0;
  height: 38px;
  line-height: 38px;
  ${({ disabled }) => {
    if (disabled) {
      return `
        opacity: 0.5;
        pointer-events: none;
      `;
    }
    return null;
  }}
`;
