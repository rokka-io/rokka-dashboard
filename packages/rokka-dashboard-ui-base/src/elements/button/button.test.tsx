import { shallow } from 'enzyme';
import React from 'react';

import { Button } from './button';

describe('Elements', () => {
  describe('Button', () => {
    it('renders a button correctly.', () => {
      const spy = jest.fn();
      const component = shallow(<Button onClick={spy}>test</Button>);

      component.simulate('click');

      expect(component).toMatchSnapshot();
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('renders a disabled button correctly and does not call onClick.', () => {
      const spy = jest.fn();
      const component = shallow(
        <Button disabled={true} onClick={spy}>
          test
        </Button>
      );

      component.simulate('click');

      expect(component).toMatchSnapshot();
      expect(spy).toHaveBeenCalledTimes(0);
    });
  });
});
