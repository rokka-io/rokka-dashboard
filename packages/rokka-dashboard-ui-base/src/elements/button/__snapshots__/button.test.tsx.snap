// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements Button renders a button correctly. 1`] = `
<styled.button
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  type="button"
>
  test
</styled.button>
`;

exports[`Elements Button renders a button without child correctly 1`] = `
<styled.button
  disabled={false}
  type="button"
/>
`;

exports[`Elements Button renders a disabled button correctly and does not call onClick. 1`] = `
<styled.button
  disabled={true}
  type="button"
>
  test
</styled.button>
`;
